db/migrate/20240705082709_create_mail_delivery_failures.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class CreateMailDeliveryFailures < ActiveRecord::Migration[7.1]
+  def change
+    create_table :mail_delivery_failures do |t|
+      t.string :email_address, null: false
+      t.string :mailer_class, null: false
+      t.string :mailer_action_method, null: false
+
+      t.timestamps
+    end
+
+    add_index :mail_delivery_failures, :email_address
```

Comment:
This works, but I think you can do `t.index :email_address` within the `create_table` block.